test/actions/pulp3/orchestration/file_sync_test.rb

```diff
@@ -41,6 +41,24 @@ def test_sync
       assert_equal repository_reference.repository_href + "versions/2/", @repo.version_href
     end
 
+    def test_sync_with_pagination
+      @repo.root.update_attributes(:url => "https://repos.fedorapeople.org/repos/pulp/pulp/fixtures/file-many/", :mirror_on_sync => false)
+      ForemanTasks.sync_task(
+          ::Actions::Pulp3::Orchestration::Repository::Update,
+          @repo,
+          @master)
+
+      SETTINGS[:katello][:pulp][:bulk_load_size] = 10
```

Comment:
this persists across tests, so i think any test that runs after this one would fail.  This would be random for each test run, so this explains all the failures you're seeing :)

I think setting it back in an ensure would probably fix it!

Reply:
i'd delete all the pulp2 vcr rerecords, as those likely will fail after this change (and maybe some of the others too)